docs/component-guides/profiles.md

```diff
@@ -1,68 +1,72 @@
+### IAM roles for service accounts
 
-The `AwsIamForServiceAccount` plugin is such a Profile plugin which allows enforcing AWS IAM access control for Profile users in order to grant/limit access to AWS resources and services.
+In order for the Profile controller to get the necessary permissions, the Profile controller pod must be recognized as an entity that can interface with AWS IAM. This is done by using [IAM roles for service accounts (IRSA)](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/).
 
-#### The Profiles Controller and the `AwsIamForServiceAccount` Plugin
+IRSA allows the use of AWS IAM permission boundaries at the Kubernetes pod level. A Kubernetes service account (SA) is associated with an IAM role with a role policy that scopes the IAM permissions (e.g. S3 read/write access, etc.). When a pod in the SA namespace is annotated with the SA name, EKS injects the IAM role ARN and a token is used to get the credentials so that the pod can make requests to AWS services within the scope of the role policy associated with the IRSA.
 
-In order for the profile controller to interface with AWS IAM access control, the profiles controller pod must be recognized as an entity that can interface with IAM and be given the necessary permissions. This is done by using IAM Role Service Accounts (IRSA).
+For more information, see [Amazon EKS IAM roles for service accounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html). 
 
-In AWS EKS, [IRSA](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/) is a feature that allows enforcing AWS IAM permissions boundaries at the Kubernetes pod level. In a nutshell, a Kubernetes service account (SA) is associated with an IAM role with a role policy that scopes the IAM permissions (e.g. S3 read/write access, etc.). When a pod in the SA namespace is annotated with the SA name, EKS injects the IAM role arn and a token used to get the credentials so that the pod can make requests to AWS services within the scope of role policy associated with the IRSA.
+### Component-level implementations
 
-In order to enforce IAM permissions boundaries at the profile level, the profiles controller IRSA itself is not given the permissions required by the profiles. Instead, admins will create an IAM role for each profile and the profile controller IRSA will have permissions to access the admin created profile role and modify it to grant the `default-editor` SA in the profile namespsace permissions to assume the role. The `default-editor` SA in the profile namespace will also be annotated with the role arn. By doing both, EKS will then inject the profile role arn and a token into the pod in the profile namespace, allowing it to receive the credentials that grant it the permissions defined in the profile role policy.
```

Comment:
What would a good title for this section be? What is the main thing that would drive customers to go looking for this information?

Reply:
Customers looking to understand how the plug works at a high level will be interested in this information, whether that's for planning an extension of the feature or understanding its security implementations. 

Prior to the README changes for the 1.4 release, this information was mainly present in code comments and was easy to loose track of.

But on the website it's probably not as important to go through all the details so here is a suggested rewording:

### Admin Considerations

- Kubeflow admins will need to create an IAM role for each profile with the desired scoped permissions
- The profile controller does not have the permissions specified in the profile roles
- The profile controller has permissions to modify the profile roles, which it will do to grant assume role permissions to the `default-editor` service account (SA) present in the profile's namespace
- A `default-editor` SA exists in every profile's namespace and will be annotated with the role arn created for the profile. Pods annotated with the SA name will be granted the Profile role permissions
- The `default-editor` SA is used by various services in Kubeflow to launch resources in profile namespaces, however, not all services do this by default

Reply:
For some reason I was thinking these changes are for the website.

I guess the info above is sufficient enough to provide context on the implementation details for the README docs.

Reply:
website is expected to render the same readme. I think the summary you put together for Admin considerations is pretty good, we should add it